### DISCUSSION TITLE
How disable pandoc manipulating abbreviations in quarto?

### DISCUSSION THREAD

**Discussion by iusgit** [quarto-dev/quarto-cli#6948] [Q&A] [answered]
### Description

Hello,
I just found out that pandoc has a list of [abbreviations](https://github.com/jgm/pandoc/blob/a19cd39a6fae3f5595c7935fbc83d4edc82300d8/data/abbreviations#L4)  where after their occurence a non-breaking space would be set in certain formats (Latex). The substitution seems to happen at the markdown reading stage.

When I search for abbreviated text (German "d. h." -> English "i.e." ) elements in my LUA filter, I wouldn't find, because there sits invisible "non-breaking space" in the string after stringify'ing and LUA's s:match and s:find cannot find "d. h." in it. It is also annoying, because in normal mode we would have `Str("d."),Space(),Str("h.")` whereas now we find `Str("d. h.")`which is not searchable.

I want to disable the abbreviation substitution, one option is to provide commandline option "--abbreviations=FILE" with an empty file. 

But how to tell Quarto to use that commandline option?

**Comment by mcanouil** [comment, marked as answer]
Without a reproducible example I can't test now.
Quarto offers `abbreviations` YAML key which take a file, see <https://quarto.org/docs/reference/formats/html.html#rendering>.
Also you can use Pandoc commandline args in Quarto, it should work: `quarto render --abbreviations=FILE`

**Reply by iusgit** [reply]
Thanks for the explanation! 

And indeed, `abbreviations` is the right keyword. I added an empty file and the NBSP went away!

**Comment by iusgit** [comment]
It is solved... But the example is:
```
---
title: "Untitled"
format: pdf
keep-tex: true
---

Test for abbreviations. e.g. Ph.D. vs. Prof. incl. Ms. Sept. is gonna get a tilda.
```
leads to tex:
```
Test for abbreviations. e.g.~Ph.D.~vs.~Prof.~incl.~Ms.~Sept.~is gonna
get a tilda.
```

But having
```
---
title: "Untitled"
format: pdf
keep-tex: true
abbreviations: abbrev.txt # empty file
---

Test for abbreviations. e.g. Ph.D. vs. Prof. incl. Ms. Sept. is gonna get a tilda.
```

we get in tex:
```
Test for abbreviations. e.g. Ph.D. vs. Prof. incl. Ms. Sept. is gonna
get a tilda.
```
Of course, those NBSP make usually sense, but if you want handle things yourself, the Pandoc AST is different than expected without disabling the abbreviations.
